cgcnn2/cli/cgcnn_ft.py

```
@@ -296,6 +306,7 @@ def main():
         valid_dataset, test_dataset = random_split(
             valid_test_dataset, lengths=[n_valid, n_test], generator=generator
         )
+        train_dataset._cache_load = functools.lru_cache(maxsize=args.cache_size)(train_dataset.__load_item)
```

Ruff (0.11.9)

Undefined name `functools`

(F821)